### PR TITLE
add moshi classes for webfinger responses

### PR DESCRIPTION
**Pull request by theScrabi**
Issue: https://github.com/owncloud/enterprise/issues/5021#issuecomment-1102692271
App: https://github.com/owncloud/android/pull/3632

### CODE REVIEW

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/webfinger/responses/WebfingerResponse.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+data class WebfingerJrdResponse (
+    val subject:String,
+    val links:List<LinkItem>
+)
+
+@JsonClass(generateAdapter = true)
+data class LinkItem(
+    val href:String,
+    val rel:String
+)
```
Comment (JuancaG05) [inline owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/webfinger/responses/WebfingerResponse.kt:39]: Better end with a blank line 👍

File: owncloudComLibrary/src/test/java/com/owncloud/android/lib/resources/webfinger/responses/WebfingerResponseTest.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        adapter.fromJson(File(fileName).readText())
+
+    @Before
+    fun prepare() {
+        val moshi = Moshi.Builder().build()
+        adapter = moshi.adapter(WebfingerJrdResponse::class.java)
+    }
+
+    @Test
+    fun `check rel in to much information - ok - correct rell is returned`() {
```
Comment (JuancaG05) [inline owncloudComLibrary/src/test/java/com/owncloud/android/lib/resources/webfinger/responses/WebfingerResponseTest.kt:24]: I don't understand very well the name of this test, what does `rel` or `rell` refer to? Is it `to` or `too`? I would change its name, don't you think?
Also, when the test is for a correct case (`ok`, like in this case), do we put additional information like `correct rell is returned` if it is evident?

File: owncloudComLibrary/src/test/java/com/owncloud/android/lib/resources/webfinger/responses/WebfingerResponseTest.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    fun `check key value pairs - ko - no rel key`() {
+        val response = loadResponses(BROKEN_JSON)!!
+        Assert.assertEquals("https://gast.somedomain.de", response.links[0].href)
+    }
+
+    companion object {
+        val RESOURCES_PATH =
+            "src/test/responses/com.owncloud.android.lib.resources.webfinger.responses"
+        val EXAMPLE_RESPONSE_JSON = "$RESOURCES_PATH/simple_response.json"
+        val TO_MUCH_INFORMATION_JSON = "$RESOURCES_PATH/to_much_information_response.json"
```
Comment (JuancaG05) [inline owncloudComLibrary/src/test/java/com/owncloud/android/lib/resources/webfinger/responses/WebfingerResponseTest.kt:46]: Same as in the test, I guess this is `TOO_...`

File: owncloudComLibrary/src/test/responses/com.owncloud.android.lib.resources.webfinger.responses/not_containing_releavant_info_response.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "subject": "acct:peter.sine@gurken.xxx",
+  "links" :
+  [
+    {
+      "rel" : "http://webfinger.example/rel/businesscard",
+      "href" : "https://www.example.com/~bob/bob.vcf"
+    }
+  ]
+}
```
Comment (JuancaG05) [inline owncloudComLibrary/src/test/responses/com.owncloud.android.lib.resources.webfinger.responses/not_containing_releavant_info_response.json:10]: Add final blank line. Also, I think the name of this file doesn't fit with the name that is used in the tests. In this case, it is called `..._releavant_...`, it should be "relevant", is that correct?

File: owncloudComLibrary/src/test/responses/com.owncloud.android.lib.resources.webfinger.responses/to_much_information_response.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      "href": "https://gast.somedomain.de",
+      "rel": "http://webfinger.owncloud/rel/server-instance"
+    },
+    {
+      "gurken": "sallat",
+      "rel" : "http://webfinger.example/rel/businesscard",
+      "href" : "https://www.example.com/~bob/bob.vcf"
+    }
+  ]
+}
```
Comment (JuancaG05) [inline owncloudComLibrary/src/test/responses/com.owncloud.android.lib.resources.webfinger.responses/to_much_information_response.json:27]: Final blank line 😀

**Review by JuancaG05 — CHANGES_REQUESTED** [review]
Some spelling and minor suggestions, good job! 👍

File: owncloudComLibrary/src/test/responses/com.owncloud.android.lib.resources.webfinger.responses/to_much_information_response.json (hunk 5 above)
Comment (abelgardep) [reply owncloudComLibrary/src/test/responses/com.owncloud.android.lib.resources.webfinger.responses/to_much_information_response.json:27]: Not relevant at all for the tests

**Review by JuancaG05 — APPROVED** [review]

**Pull request merged**